runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/DispatchAgent.java

```diff
@@ -434,6 +438,37 @@ public void detachSender(
         throttles[throttleIndex(streamId)].remove(instanceId(streamId));
     }
 
+    @Override
+    public void detachStreams(
+        long bindingId)
+    {
+        IntHashSet streamIdSet = streamIndicesByBinding.remove(bindingId);
+        if (streamIdSet != null)
+        {
+            streamIdSet.forEach(streamId ->
+                {
+                    streams[streamId].values().forEach(streamHandler ->
+                        doSyntheticAbort(streamId, streamHandler)
+                    );
+                    streams[streamId] = new Int2ObjectHashMap<>();
+                }
+            );
+        }
+
+        IntHashSet throttleIdSet = throttleIndicesByBinding.remove(bindingId);
+        if (throttleIdSet != null)
+        {
+            throttleIdSet.forEach(throttleId ->
+                {
+                    throttles[throttleId].values().forEach(streamHandler ->
+                        doSyntheticAbort(throttleId, streamHandler)
+                    );
+                    throttles[throttleId] = new Int2ObjectHashMap<>();
+                }
+            );
+        }
+    }
+
```

Comment:
Please reevaluate this implementation after applying feedback for `handleBeginInitial` below.